indexer/indexer_tests/osm_editor_test.cpp

```diff
@@ -577,35 +672,363 @@ void EditorTest::HaveMapEditsToUploadTest()
+  });
+
+  ForEachCafeAtPoint(m_index, m2::PointD(2.0, 2.0), [&editor, &createAndCheckNote](FeatureType & ft)
+  {
+    createAndCheckNote(ft.GetID(), {2.0, 2.0}, osm::Editor::NoteProblemType::General);
+
+    TEST_NOT_EQUAL(editor.GetFeatureStatus(ft.GetID()), osm::Editor::FeatureStatus::Obsolete, ());
+    auto notes = editor.m_notes->GetNotes();
+    TEST_EQUAL(notes.front().m_note.find(osm::Editor::kPlaceDoesNotExistMessage), string::npos, ());
+  });
+}
+
+void EditorTest::LoadMapEditsTest()
+{
+  auto & editor = osm::Editor::Instance();
+
+  auto const gbMwmId = BuildMwm("GB", [](TestMwmBuilder & builder)
+  {
+    builder.Add(TestCafe(m2::PointD(0.0, 0.0), "London Cafe", "en"));
+    builder.Add(TestCafe(m2::PointD(1.0, 1.0), "London Cafe", "en"));
+  });
+
+  auto const rfMwmId = BuildMwm("RF", [](TestMwmBuilder & builder)
+  {
+    builder.Add(TestCafe(m2::PointD(2.0, 2.0), "Moscow Cafe1", "en"));
+    builder.Add(TestCafe(m2::PointD(7.0, 7.0), "Moscow Cafe2", "en"));
+    builder.Add(TestCafe(m2::PointD(4.0, 4.0), "Moscow Cafe3", "en"));
+    builder.Add(TestCafe(m2::PointD(6.0, 6.0), "Moscow Cafe4", "en"));
+  });
+
+  vector<FeatureID> features;
+
+  ForEachCafeAtPoint(m_index, m2::PointD(0.0, 0.0), [&features](FeatureType & ft)
+  {
+    SetBuildingLevelsToOne(ft);
+    features.emplace_back(ft.GetID());
+  });
+
+  ForEachCafeAtPoint(m_index, m2::PointD(1.0, 1.0), [&editor, &features](FeatureType & ft)
+  {
+    editor.DeleteFeature(ft.GetID());
+    features.emplace_back(ft.GetID());
+  });
+
+  ForEachCafeAtPoint(m_index, m2::PointD(2.0, 2.0), [&editor, &features](FeatureType & ft)
+  {
+    editor.MarkFeatureAsObsolete(ft.GetID());
+    features.emplace_back(ft.GetID());
+  });
+
+  ForEachCafeAtPoint(m_index, m2::PointD(7.0, 7.0), [&features](FeatureType & ft)
+  {
+    SetBuildingLevelsToOne(ft);
+    features.emplace_back(ft.GetID());
+  });
+
+  ForEachCafeAtPoint(m_index, m2::PointD(4.0, 4.0), [&editor, &features](FeatureType & ft)
+  {
+    editor.DeleteFeature(ft.GetID());
+    features.emplace_back(ft.GetID());
+  });
+  
+  ForEachCafeAtPoint(m_index, m2::PointD(6.0, 6.0), [&features](FeatureType & ft)
+  {
+    SetBuildingLevelsToOne(ft);
+    features.emplace_back(ft.GetID());
+  });
+
+  osm::EditableMapObject emo;
+  CreateCafeAtPoint({5.0, 5.0}, rfMwmId, emo);
+  features.emplace_back(emo.GetID());
+
+  editor.Save();
+  editor.LoadMapEdits();
+
+  auto const fillLoaded = [&editor](vector<FeatureID> & loadedFeatures)
+  {
+    loadedFeatures.clear();
+    for (auto const & mwm : editor.m_features)
+    {
+      for (auto const & index : mwm.second)
+      {
+        loadedFeatures.emplace_back(index.second.m_feature.GetID());
+      }
+    }
+  };
+
+  vector<FeatureID> loadedFeatures;
+  fillLoaded(loadedFeatures);
+
+  sort(loadedFeatures.begin(), loadedFeatures.end());
+  sort(features.begin(), features.end());
+  TEST_EQUAL(features, loadedFeatures, ());
+
+  auto const removeLastMwm = [this]()
```

Comment:
It would be more readable to remove mwm by id or by name. Also, to simplify test code, please, extract this into an individual method.